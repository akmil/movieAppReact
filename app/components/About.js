import * as React from 'react';

export class About extends React.Component {

    render() {
        return (
            <p>About tab: {this.props.match.params.tab}, and section: {this.props.match.params.section}</p>
        );
    }
}