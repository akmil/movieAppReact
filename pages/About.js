import * as React from 'react';

export class About extends React.Component {
  getInitialProps(ctx) {
    // component will read it from store's state when rendered
    console.log('About.getInitialProps ->>', ctx, this.props);
    return {};
  }

  constructor(props) {
    super(props);
    console.log('constructor props(About)', props);
  }

  render() {
    return (
        <p>About tab:</p>
    );
  }
}

export default About;
