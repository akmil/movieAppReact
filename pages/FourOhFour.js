import * as React from 'react';
import Search from "./search";

export class FourOhFour extends React.Component {

    render() {
        return (
            <p>404 Not found</p>
        );
    }
}

export default FourOhFour;