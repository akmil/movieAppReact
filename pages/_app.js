import React from 'react';
import { Provider } from 'react-redux';
import App, { Container } from 'next/app';
import withRedux from 'next-redux-wrapper';

import configureStore from '../app/r_store/configureStore.BE';

class MyApp extends App {
  render() {
    const { Component, pageProps, store } = this.props;

    console.log('this.props MyApp', this.props);

    return (
            <Container>
                <Provider store={store}>
                    <Component {...pageProps} />
                </Provider>
            </Container>
    );
  }
}

export default withRedux(configureStore)(MyApp);
