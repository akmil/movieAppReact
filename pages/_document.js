import Document, { Head, Main, NextScript } from 'next/document';
import stylesheet from 'styles/main.scss';

export default class MyDocument extends Document {
  render() {
    return (
            <html>
            <Head>
                <link rel="stylesheet" href="/_next/static/main.css" />
                <style dangerouslySetInnerHTML={{ __html: stylesheet }} />
            </Head>
            <body>
            <Main />
            <NextScript />
            </body>
            </html>
    );
  }
}
