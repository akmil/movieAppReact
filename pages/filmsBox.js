import React from 'react';

import {FilmsTitle}         from '../app/components/film/filmsParts/filmsTitle';
import {FilmsCover}         from '../app/components/film/filmsParts/filmsCover';
import {FilmsDescription}   from '../app/components/film/filmsParts/filmsDescription';
import {FilmsList}          from '../app/components/film/filmsParts/filmsList';
import FilmsSearchButton    from '../app/components/film/filmsParts/filmsSearchButton';

import * as actions from "../app/r_actions";
import {connect} from "react-redux";

class FilmsBox extends React.Component {
    static getInitialProps = (ctx) => {
        const { id } = ctx.query;
        console.log('ctx', ctx);
        return {
            id,
        }
    };

    constructor(props){
        super(props);
        this.state = {detailMovie: {title:''}}
        console.log('FilmsBox', this.props);
        // this.handleInputChange = this.handleInputChange.bind(this);
        // this.handleFormSubmit = this.handleFormSubmit.bind(this);
    }

    componentDidMount(){
        this.props.fetchMovieById(this.props.id)
            .then(res => {
                this.setState({detailMovie : res});

                //move to componentDidUpdate
                this.props.fetchMovieGenre(res.genres[0])
                    .then(res => {
                        this.setState({relatedMovies : res});
                        console.log(this.state, res);
                    })
            })
    }

    render() {
        return (
            <div className='films films-box-wrapper'>
                <p>films-box: <b>{this.props.id}</b></p>
                <div className='films-box-header'>
                    <div className='logo col-33'>netfixroulette</div>
                    <FilmsSearchButton name={'FilmsSearchButton'} theme={'white'}/>
                    <div className={'d-flex'}>
                        <FilmsCover src={this.state.detailMovie.poster_path}/>
                        <div className='films-box-body'>
                            <FilmsTitle activeClass={'activeClass'}
                                        title={this.state.detailMovie.title}
                                        rate={this.state.detailMovie.vote_average}
                            />
                            <div className='mb-3'>
                                <span className='mr-3 release'>{this.state.detailMovie.release_date}</span>
                                <span className='ml-3 duration'>{this.state.detailMovie.runtime}</span>
                            </div>
                            <FilmsDescription
                                description={this.state.detailMovie.overview}
                            />
                        </div>
                    </div>
                </div>
                <div className='results results-wrapper'>
                    {/*<FilmsSortBy
                        filter={'SHOW_DRAMA'}
                        name = {'show Drama'}
                        theme={'white'}
                    />

                    <FilmsSortBy
                        theme={'white'}
                        filter={'SHOW_COMEDY'}
                        name = {'show Comedy'}
                    />
                    <FilmsSortBy
                        theme={'white'}
                        filter={'SHOW_COMEDY'}
                        name = {'fetch'}
                    />*/}
                </div>
                <FilmsList></FilmsList>
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    // console.log('ownProps', ownProps, state);
    return {
        sortBy: state.sortBy,
        totalMovies: state.movies.length
    }
};

const mapDispatchToProps = ({
    fetchMovieById: actions.fetchMovieById,
    fetchMovieGenre: actions.fetchMovieGenre
})

export default connect(mapStateToProps, mapDispatchToProps)(FilmsBox);