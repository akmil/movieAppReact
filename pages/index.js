import React from 'react';
import Header from '../app/components/header/header';
import ResultsBox from '../app/components/resultsBox/resultsBox';


class HomeLayout extends React.Component {
  // static getInitialProps = async ({ reduxStore, req, store })=> {
  //     const isServer = !!req;
  //     console.log('HomeLayout ->>', store, isServer);
  //     await store.dispatch({type: 'TEST' });
  //
  //     return {};
  // }

  constructor(props) {
    super(props);
    // console.log('props(HomeLayout)', props);
  }

  componentDidMount() {
    // console.log('HomeLayout', this.props.movies);
  }

  componentWillUnmount() {
    // clearInterval(this.timer)
  }

  render() {
    return (
            <div className={'main-wrapper'}>
                <Header></Header>
                <ResultsBox></ResultsBox>
            </div>
    );
  }
}


// export default connect()(withRouter(HomeLayout));
export default HomeLayout;
// export default HomeLayout;
