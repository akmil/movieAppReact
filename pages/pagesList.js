import React from 'react';
import { Link } from 'next/link';

class PagesList extends React.Component {
  render() {
    return (
            <div >
                <ul>
                    <li><Link href="/home">Home</Link></li>
                    <li><Link href="/details/movieName">Details</Link></li>
                    <li><Link href="/about/company/employees">About</Link></li>
                    <li><Link href="/workers">Workers(redirect)</Link></li>
                    <li><Link href="/some-page">goto 404</Link></li>
                </ul>

                {this.props.children}
            </div>
    );
  }
}

export default PagesList;
