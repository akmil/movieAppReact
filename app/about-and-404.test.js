import React from 'react';
// import ReactDOM from 'react-dom';
// import renderer from 'react-test-renderer';
import {About} from "../pages/About";
import {FourOhFour} from "../pages/FourOhFour";
import { shallow } from 'enzyme'
import PagesList from "./pagesList";
import Search from "./search";

describe('About, FourOhFour, PagesList, Search', function () {
    it('should render About without errors', () => {
      // const wrapper = renderer.create(<About/>)
        const wrapper = shallow(<About match={ {params:{tab:"tabName", section: 'sectionTestName'}} }/>);
        expect(wrapper).toMatchSnapshot();
    })

    it('should render FourOhFour without errors', () => {
      // const wrapper = renderer.create(<FourOhFour/>)
        const wrapper = shallow(<FourOhFour/>);
        expect(wrapper).toMatchSnapshot();
    })

    it('should render PagesList without errors', () => {
      // const wrapper = renderer.create(<FourOhFour/>)
        const wrapper = shallow(<PagesList/>);
        expect(wrapper).toMatchSnapshot();
    })

    it('should render Search without errors', () => {
      // const wrapper = renderer.create(<FourOhFour/>)
        const wrapper = shallow(
            <Search match={ {params:{query:"querySearch"}} }/>
        );
        expect(wrapper).toMatchSnapshot();
    })

});
