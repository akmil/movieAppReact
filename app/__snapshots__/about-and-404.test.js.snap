// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About, FourOhFour, PagesList, Search should render About without errors 1`] = `
<p>
  About tab:
</p>
`;

exports[`About, FourOhFour, PagesList, Search should render FourOhFour without errors 1`] = `
<p>
  404 Not found
</p>
`;

exports[`About, FourOhFour, PagesList, Search should render PagesList without errors 1`] = `
<div>
  <ul />
</div>
`;

exports[`About, FourOhFour, PagesList, Search should render Search without errors 1`] = `
<div
  className="main-layout-wrapper"
>
  <Connect(withRouter(Header)) />
  <ResultsBox />
</div>
`;
